### DISCUSSION TITLE
Rules - allow InsertAction to insert empty values

### DISCUSSION THREAD

**Discussion by oba-reliatec** [reliatec-gmbh/LibreClinica#393] [Ideas]
**Description:**
With rules InsertAction, it is possible to enter values in CRF fields (items). However, it is not possible to insert empty values "", and thus delete old entries.

**Requirements:**
InsertAction rule with empty value

**Steps to follow:**
1. login (you might use [this](https://www.reliatec.de/en/branchen-loesungen/medizin-it/klinische-studien/libreclinica/demo) instance, which provides a CRF matching the sample rule below)
2. select Build Study from Tasks menu and click add rule (+)
3. upload this [sample rule](https://github.com/reliatec-gmbh/LibreClinica/files/14839055/InsertAction_empty.xml.zip) (make sure to unzip first, GitHub does not support XML files for upload)

**Expected result:**
Rule is uploaded without error

**Actual result:**
An error message appears: InsertAction is not valid: value.empty

**Comment by toskrip** [comment]
Did not checked all the conditions but in general the insert action finishes by creating new ItemData (if does not exist) or modifying existing ItemData (if exists) with provided insert action value. I think the main reason why empty value is currently invalid is that one would want to prevent creating new ItemData with empty value at the first place. Modifying existing with empty value I think could be useful. We will need to examine it properly to ensure that validation logic is still followed.
